include/deal.II/lac/petsc_vector_base.h

```diff
@@ -523,19 +535,21 @@ namespace PETScWrappers
      *
      * For complex valued vector, this gives$\left(v^\ast,vec\right)$.
      */
-    PetscScalar operator*(const VectorBase &vec) const;
+    virtual PetscScalar
+    operator*(const VectorSpaceVector<PetscScalar> &vec) const override;
 
     /**
      * Return the square of the $l_2$-norm.
      */
     real_type
-    norm_sqr() const;
+    norm_sqr()
+      const; // TODO - should we deprecate? this is not in VectorSpaceVector
```

Comment:
yes.

Reply:
`LA::D::Vector` implements this so I kept it.